features/events/api/src/main/java/org/opennms/netmgt/xml/event/AlarmData.java

```diff
@@ -116,6 +119,24 @@ public AlarmData() {
         super();
     }
 
+    public static AlarmData copyFrom(IAlarmData source) {
+        if (source == null) {
+            return null;
+        }
+
+        AlarmData alarmData = new AlarmData();
+        alarmData.setReductionKey(source.getReductionKey());
+        alarmData.setAlarmType(source.getAlarmType());
+        alarmData.setClearKey(source.getClearKey());
+        alarmData.setAutoClean(source.getAutoClean());
+        alarmData.setX733AlarmType(source.getX733AlarmType());
+        alarmData.setX733ProbableCause(source.getX733ProbableCause());
+        alarmData.getUpdateFieldList().addAll(
+                source.getUpdateFieldList().stream().map(UpdateField::copyFrom).collect(Collectors.toList()));
+        alarmData.setManagedObject(
+                source.getManagedObject() == null ? null : ManagedObject.copyFrom(source.getManagedObject()));
```

Comment:
Minor nitpick. The null check here isn't necessary since the `copyFrom()` methods already handle the null case.

Reply:
Done, removed the unnecessary null checks.